benchmarks/torchbench/benchfile.py

```diff
@@ -107,18 +107,27 @@ def pin(self, *pip_compile_args, constraints:list=tuple()):
             self._clone_tb(pin_dir)
 
             (self.pack_path / "reqs").copy(pin_dir / "reqs")
-            (self.pack_path / "benchtest.yaml").copy(pin_dir)
             super().pin(*pip_compile_args, requirements_file=req_file,
                         input_files=(req_file.with_suffix('.in'),), constraints=constraints,
                         cwd=pin_dir)
             (pin_dir / "reqs").merge_into(self.pack_path / "reqs")
 
     def exec_pip_compile(self, requirements_file:XPath, input_files:list,
                          *pip_compile_args, cwd:XPath):
-        self.execute(cwd / "reqs/pip-compile.sh", "--reqs", "reqs",
-            "--tb-root", ".", "--", "--resolver", "backtracking",
-            "--output-file", requirements_file, *pip_compile_args, *input_files,
-            cwd=cwd)
+        config = parse_config(self.config["config_file"])
+        models = set()
+        for name, defn in config["benchmarks"].items():
+            group = defn.get("group", name)
+            if group == self.config["group"]:
+                models.add(defn["model"])
+        models_args = []
+        for m in models:
+            models_args.extend(("-m", m))
+        self.execute(cwd / "reqs/pip-compile.sh", "--reqs", "reqs", "--tb-root", ".",
+                     "--config", (self.pack_path / "benchtest.yaml").absolute(),
+                     *models_args, "--", "--resolver", "backtracking",
+                     "--output-file", requirements_file, *pip_compile_args,
+                     *input_files, cwd=cwd)
```

Comment:
Lmiting the number of models here by using only the models from the config group could help managing the dependencies as we don't try to be compatible with the benches we don't use but the result of the pin would also be different if from a config file to the other